Applications/WebPortal/WebApp/library/source/pages/account_page/compliance_page/compliance_page.tsx

```diff
@@ -43,36 +51,57 @@ export class CompliancePage extends React.Component<Properties, State> {
       <div style={CompliancePage.STYLE.wrapper}>
         <div style={CompliancePage.STYLE.filler}/>
         <div style={contentStyle}>
-          <RulesList 
+          <RulesList
             displaySize={this.props.displaySize}
             currencyDatabase={this.props.currencyDatabase}
             complianceList={this.state.complianceList}
-            onChange={this.onChange}/>
+            onChange={this.onRuleChange}/>
           <div style={CompliancePage.STYLE.paddingMedium}/>
-          <NewRuleButton displaySize={this.props.displaySize}/>
+          <NewRuleButton displaySize={this.props.displaySize}
+            isOpen={this.state.isAddRulewModalOpen}
+            onToggleModal={this.onToggleAddRuleModal}
+            onAddNewRule={this.onAddNewRule}
+            ruleSchemas={this.props.ruleSchemas}/>
           <div style={CompliancePage.STYLE.paddingLarge}/>
         </div>
         <div style={CompliancePage.STYLE.filler}/>
       </div>);
   }
 
-  private onChange(ruleIndex: number, newRule: Nexus.ComplianceRuleEntry) {
+  private onRuleChange(ruleIndex: number, newRule: Nexus.ComplianceRuleEntry) {
     this.state.complianceList[ruleIndex] = newRule;
     this.setState({complianceList: this.state.complianceList});
   }
 
+  private onToggleAddRuleModal() {
+    this.setState({isAddRulewModalOpen: !this.state.isAddRulewModalOpen});
+  }
+
+  private onAddNewRule(schema: Nexus.ComplianceRuleSchema) {
+    this.state.complianceList.push(
```

Comment:
Good attempt but unfortunately not the route to go in. Instead this page will have to behave like a "controlled" component. When a new rule is added it invokes a callback prop (you can call it onRuleAdded).

Reply:
Could we meet up shortly to discuss this? I have a concern about how it's going to affect how rules are modified.